isis/src/base/apps/cubeatt/cubeatt.cpp

```diff
@@ -0,0 +1,56 @@
+#include "cubeatt.h"
+
+#include "ProcessByLine.h"
+
+using namespace std;
+using namespace Isis;
+
+namespace Isis {
+  void cubeattProcess(Buffer &in, Buffer &out);
+
+  void cubeatt(UserInterface &ui) {
+    Cube icube;
+    CubeAttributeInput inAtt = ui.GetInputAttribute("FROM");
+    if (inAtt.bands().size() != 0) {
+      icube.setVirtualBands(inAtt.bands());
+    }
+    icube.open(ui.GetFileName("FROM"));
+    cubeatt(&icube, ui);
+  }
+
+
+  void cubeatt(Cube *icube, UserInterface &ui) {
+    bool propTables = ui.GetBoolean("PROPTABLES");
+    QString outputFileName = ui.GetFileName("TO");
+    CubeAttributeOutput outputAttributes= ui.GetOutputAttribute("TO");
+    cubeatt(icube, outputFileName, outputAttributes, propTables);
+  }
+
+
+  void cubeatt(Cube *icube, QString outputCubePath, CubeAttributeOutput outAttributes, bool propTables) {
+    // We will be processing by line
+    ProcessByLine p;
+
+    // Should we propagate tables
+    if(propTables) {
```

Comment:
This seems backwards.

Reply:
Good catch! Also, this is weirdly written anyway. I've cleaned it up now to just do: `p.PropagateTables(propTables);`